### PR TITLE
implement DeriveIden in sea-orm only

### PR DESCRIPTION
**Pull request by darkmmon**
## PR Info

- Closes https://github.com/SeaQL/sea-orm/issues/1473

- Dependencies:
  - 

- Dependents:
  - 

## New Features

- [ ]  Using DeriveIden now does not require dependency on sea-query

## Bug Fixes

- [ ] 

## Breaking Changes

- [ ]  DeriveIden now has a different implementation.

## Changes

- [ ]

### CODE REVIEW

File: tests/derive_iden_tests.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+fn main() -> Result<(), DbErr> {
+    assert_eq!(Class::Id.to_string(), "id");
+    assert_eq!(Class::Title.to_string(), "title");
+    assert_eq!(Class::Text.to_string(), "text");
+
+    assert_eq!(Glyph.to_string(), "glyph");
+
+    assert_eq!(Book::Id.to_string(), "id");
+    assert_eq!(Book::Title.to_string(), "turtle");
+    assert_eq!(Book::Text.to_string(), "te_xt");
```
Comment (billy1624) [inline tests/derive_iden_tests.rs:35]: I think this should be...?
```suggestion
    assert_eq!(Book::Text.to_string(), "TeXt");
```

File: sea-orm-macros/src/derives/derive_iden.rs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            let var_ident = &v.ident;
+            let mut var_name = var_ident.to_string().to_snake_case();
+            v.attrs
+                .iter()
+                .filter(|attr| attr.path().is_ident("sea_orm"))
+                .try_for_each(|attr| {
+                    attr.parse_nested_meta(|meta| {
+                        if meta.path.is_ident("iden") {
+                            let litstr: LitStr = meta.value()?.parse()?;
+                            var_name = litstr.value().to_snake_case();
```
Comment (billy1624) [inline sea-orm-macros/src/derives/derive_iden.rs:60]: I guess we need to edit this line

File: tests/derive_iden_tests.rs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+#[derive(DeriveIden)]
+pub enum Class {
+    Id,
+    Title,
+    Text,
+}
+
+#[derive(DeriveIden)]
+struct Glyph;
```
Comment (billy1624) [inline tests/derive_iden_tests.rs:14]: Can we also test renaming the iden of unit struct?
```suggestion
#[derive(DeriveIden)]
#[sea_orm(iden = "_glyph")]
struct Glyph;
```

Comment (billy1624) [reply tests/derive_iden_tests.rs:14]: We should create a new struct to test it

File: tests/derive_iden_tests.rs (hunk 1 above)
Comment (darkmmon) [reply tests/derive_iden_tests.rs:35]: I think Iden in SeaQuery would make the name to be in snake case, no matter if it is from rename or the original name itself?
Having different behaviour from SeaQuery and SeaORM seems weird.

Comment (darkmmon) [reply tests/derive_iden_tests.rs:35]: oh wait 
maybe I was misreading the SeaQuery codes...
Seems like it doesn't change it after I re-read the code in SeaQuery.
Will change it back.

**Comment by tyt2y3** [comment]
I want to make sea-query-derive an optional dependency. I will do it in another PR

File: tests/derive_iden_tests.rs (hunk 1 above)
Comment (billy1624) [reply tests/derive_iden_tests.rs:35]: Please correct me if I'm wrong. The original behaviour of `sea_query::Iden` should be...

```rs
#[derive(Iden)]
#[iden = "_glyphAbcD"]
struct Glyph;

assert_eq!(Glyph.to_string(), "_glyphAbcD");
```

**Comment by plasticbox** [comment]
Hi. I know you guys are working on this.
How long do you think it will take to deploy?
I can't build now... If it takes time to deploy, is there a way to bypass it?

I think need a clear policy on versions and dependencies.

**Comment by billy1624** [comment]
Hey @plasticbox, as a temporary workaround, please import the `sea_orm::sea_query` into scope.

```rs
use sea_orm::sea_query::{self};
```

**Comment by plasticbox** [comment]
> Hey @plasticbox, as a temporary workaround, please import the `sea_orm::sea_query` into scope.
> 
> ```rust
> use sea_orm::sea_query::{self};
> ```

I put that in table #[derive(Iden)].
But still I get this message and it won't build.

image

**Comment by plasticbox** [comment]
> Hey @plasticbox, as a temporary workaround, please import the `sea_orm::sea_query` into scope.
> 
> ```rust
> use sea_orm::sea_query::{self};
> ```

[dependencies.sea-orm-migration]
version = "0.11.0"

[dependencies.sea-orm-migration]
version = "~0.10.1"

I lowered the sea-orm-migration version to 0.10 and the build returned to normal.
I disabled the FunctionCall usage code that was added in version 0.11.

**Comment by tyt2y3** [comment]
@plasticbox I believe the build error you saw has been resolved now. If not please open a new thread with details of the error.

File: tests/derive_iden_tests.rs
Comment (tyt2y3) [inline tests/derive_iden_tests.rs:1]: This is lovely.

**Pull request merged**